ops/ccs-ops-misc/load_test/common/stats.py

```diff
@@ -0,0 +1,274 @@
+            arg_str (str): The "arg string" representation of this object.
+
+        Raises:
+            ValueError: Raised if the passed string does not follow the proper format.
+
+        Returns:
+            StatsStorageConfig: Returns a concrete instance of StatsStorageConfig with the values specified in the "arg string".
+        """
+        items = arg_str.split(':')
+        if len(items) != 4:
+            raise ValueError(
+                'Input must follow format: "<STORAGE_TYPE>:<RUNNING_ENVIRONMENT>:<TAG>:<PATH_OR_BUCKET>"') from None
+
+        try:
+            storage_type = StatsStorageType[items[0].upper()]
+        except KeyError:
+            raise ValueError(
+                'STORAGE_TYPE must be either "file" or "s3"') from None
+
+        try:
+            stats_environment = StatsEnvironment[items[1].upper()]
+        except KeyError:
+            raise ValueError(
+                'RUNNING_ENVIRONMENT must be either "TEST" or "PROD"') from None
+
+        tag = items[2]
+        # Tag must follow the BFD Insights data convention constraints for
+        # partition/folders names, as it is used as a partition folder when uploading
+        # to S3
+        if re.fullmatch('[a-z0-9_]+', tag) == None:
+            raise ValueError(
+                'TAG must only consist of lower-case letters, numbers and the "_" character') from None
+
+        if storage_type == StatsStorageType.FILE:
+            return StatsFileStorageConfig(stats_environment, tag, items[3])
+        else:
+            return StatsS3StorageConfig(stats_environment, tag, items[3])
+
+
+@dataclass
+class StatsFileStorageConfig(StatsStorageConfig):
+    """Concrete dataclass inheriting from StatsStorageConfig distinguishing a config for storing
+    statistics files to a local file"""
+    file_path: str
+    """The parent path of the statistics file that will be written to disk"""
+
+    def to_arg_str(self) -> str:
+        return f'file:{self.stats_environment.name}:{self.tag}:{self.file_path}'
+
+
+@dataclass
+class StatsS3StorageConfig(StatsStorageConfig):
+    """Concrete dataclass inheriting from StatsStorageConfig distinguishing a config for storing
+    statistics files to an S3 Bucket"""
+    bucket: str
+    """The AWS S3 Bucket that statistics will be written to"""
+
+    def to_arg_str(self) -> str:
+        return f's3:{self.stats_environment.name}:{self.tag}:{self.bucket}'
+
+
+class AggregatedStats(object):
+    """Represents a snapshot of aggregated performance statistics of all tasks, or endpoints, that
+    ran in the current Locust environment"""
+
+    def __init__(self, locust_env: Environment, percentiles_to_report: List[float], stats_tag: str, running_env: StatsEnvironment = StatsEnvironment.TEST) -> None:
+        """Creates a new instance of AggregatedStats given the current Locust environment and a list of percentiles to report.
+
+        Args:
+            environment (Environment): Current Locust environment
```

Comment:
super-nit: Parameter is named `locust_env`

Reply:
Ah, I didn't catch that one when I refactored that parameter. I'll update it to `locust_env`.